### ISSUE TITLE
Improve Graphql endpoint to support dynamic schemas

### ISSUE THREAD

**Issue by cesarvarela** [responsible-ai-collaborative/aiid#2256]
With the advent of #2047, we must perform complex queries on the `classifications` collection and support dynamic graphQL schemas built from the definitions found in the `taxas` collection.

## Current Architecture
 
<img width="756" alt="image" src="https://github.com/responsible-ai-collaborative/aiid/assets/1172479/82c606a2-d401-42be-ae16-47fc495fde65">

### Classifications collection schema
<img width="759" alt="image" src="https://github.com/responsible-ai-collaborative/aiid/assets/1172479/b0b1efbb-b316-4756-b2f0-52b56fe6bff9">

## Update Option 1

<img width="716" alt="image" src="https://github.com/responsible-ai-collaborative/aiid/assets/1172479/a7f44358-ee9d-42fb-a187-f1cf194c4c6d">

## Update Option 2

<img width="869" alt="image" src="https://github.com/responsible-ai-collaborative/aiid/assets/1172479/d954b8b1-c7b9-48d2-8c49-885f25673f13">

**Comment by cesarvarela** [comment]
> Code lives in Atlas App Services, not GitHub. This makes it more difficult to test and maintain.

Although code is run there, it is being version controlled: https://github.com/responsible-ai-collaborative/aiid/tree/master/site/realm and deployed via https://github.com/responsible-ai-collaborative/aiid/blob/master/.github/workflows/realm-production.yml

We also have some unit tests for them: https://github.com/responsible-ai-collaborative/aiid/tree/master/site/gatsby-site/cypress/e2e/unit/functions

@kepae

**Comment by kepae** [comment]
Good point, thanks for the definitive correction on that -- we already maintain, test, and push code to realm/app services here in the repo.

**Comment by cesarvarela** [comment]
From yesterday's discussion:

## Conclusions

- We have to run queries against a dataset that will get increasingly bigger
- We are storing classifications in MongoDB so we have no other option (at least without Rube Goldberging this) than to use Mongo's query engine
- To properly use the query engine, we have to store classification attributes with the correct data type instead of using the `JSON.stringify` value

## Actions

- Add a `value` field to classification attributes by creating a migration that stores the `JSON.parse`d value from the `value_json` attribute
- Create a custom resolver that takes care of classification upserts so we always keep a `value` field up to date
- Create a custom resolver that receives a MongoDB query expression, forwards it to the query engine (`collection.find`), and returns the results

Once this is done,  we can go back to #2047
